becquerel/core/fitting.py

```diff
@@ -0,0 +1,830 @@
+                    title_fontweight='bold', residuals='abs', **kwargs):
+        ymin, ymax = self.y_roi.min(), self.y_roi.max()
+        # Prepare plots
+        dx, dx_roi = self.dx, self.dx_roi
+        if dx is None:
+            dx = np.ones_like(self.x)
+        if dx_roi is None:
+            dx_roi = np.ones_like(self.x_roi)
+        gs = GridSpec(2, 2, height_ratios=(4, 1))
+        gs.update(left=0.05, right=0.99, wspace=0.03, top=0.94, bottom=0.06,
+                  hspace=0.06)
+        fig = plt.figure(figsize=(18, 9))
+        fit_ax = fig.add_subplot(gs[0, 0])
+        res_ax = fig.add_subplot(gs[1, 0], sharex=fit_ax)
+        txt_ax = fig.add_subplot(gs[:, 1])
+        # Set fig title
+        if title is not None:
+            fig.suptitle(str(title), fontweight=title_fontweight,
+                         fontsize=title_fontsize)
+
+        # ---------------------------------------
+        # Fit plot (keep track of min/max in roi)
+        # ---------------------------------------
+        # Smooth roi x values
+        x_plot = np.linspace(self.x_roi[0], self.x_roi[-1], 1000)
+        # All data (not only roi)
+        fit_ax.errorbar(self.x, self.y/dx, yerr=self.y_unc,
+                        c='k', fmt='o', markersize=5, alpha=0.1, label='data')
+        # Init fit
+        y = self.eval(x_plot, **self.result.init_values)
+        ymin, ymax = min(y.min(), ymin), max(y.max(), ymax)
+        fit_ax.plot(x_plot, y, 'k--', label='init')
+        # Best fit
+        y = self.eval(x_plot, **self.result.best_values)
+        ymin, ymax = min(y.min(), ymin), max(y.max(), ymax)
+        fit_ax.plot(x_plot, y, color='#e31a1c', label='best fit')
+        # Components (currently will work for <=3 component)
+        colors = ['#1f78b4', '#33a02c', '#6a3d9a']
+        for i, m in enumerate(self.result.model.components):
+            y = m.eval(x=x_plot, **self.result.best_values)
+            if isinstance(y, float):
+                y = np.ones(x_plot.shape) * y
+            ymin, ymax = min(y.min(), ymin), max(y.max(), ymax)
+            fit_ax.plot(x_plot, y, label=m.prefix, color=colors[i])
+        # Plot Peak center and FWHM
+        peak_centers = [self.param_val(p) for p in self.param_names if
+                        (p.startswith('gauss') and p.endswith('mu'))]
+        peak_fwhms = [self.param_val(p) * FWHM_SIG_RATIO
+                      for p in self.param_names if
+                      (p.startswith('gauss') and p.endswith('sigma'))]
+        for i, (c, f) in enumerate(zip(peak_centers, peak_fwhms)):
+            if i == 0:
+                label = 'Centroid and FWHM'
+            else:
+                label = None
+            fit_ax.axvline(c, color='#ff7f00')
+            fit_ax.axvspan(c - f / 2.0, c + f / 2.0, color='#ff7f00',
+                           alpha=0.2, label=label)
+        # Misc
+        fit_ax.legend(loc='upper right')
+        # TODO: add ylabel based on units of y
```

Comment:
Should this be an issue?

Reply:
I typically default to 'counts per bin' and update it later with something like
```py
ax.set_ylabel('counts per {0:.1f} keV'.format(binwidths[0]))
```
I guess here we can get the units from `Spectrum`, too. A new issue for this makes sense to me.

Reply:
Leave this alone for now and address in #224